store/src/main/java/org/apache/rocketmq/store/index/IndexService.java

```diff
@@ -366,7 +366,18 @@ public void start() {
 
     }
 
+    // fix issue 4463
     public void shutdown() {
-
+        if (!indexFileList.isEmpty()) {
+            final IndexFile indexFile = indexFileList.get(indexFileList.size() - 1);
+            long indexMsgTimestamp = 0;
+            if (indexFile.isWriteFull()) {
+                indexMsgTimestamp = indexFile.getEndTimestamp();
+            }
+            indexFile.flush();
+            if (indexMsgTimestamp > 0) {
+                this.defaultMessageStore.getStoreCheckpoint().setIndexMsgTimestamp(indexMsgTimestamp);
+            }
```

Comment:
Could we just call `flush(indexFile)` here?

Reply:
I don't want to call getstorecheckpoint().flush() twice during shutdown .This code can be optimized, but the flush method should be modified

Reply:
because class DefaultMessageStore will call  storeCheckpoint.flush() after indexService.shutdown()

    this.storeStatsService.shutdown();
    this.indexService.shutdown();
    this.commitLog.shutdown();
    this.reputMessageService.shutdown();
    this.flushConsumeQueueService.shutdown();
    this.allocateMappedFileService.shutdown();
    this.storeCheckpoint.flush();
    this.storeCheckpoint.shutdown();